pulp_rpm/app/tasks/publishing.py

```diff
@@ -156,28 +161,53 @@ def populate(self):
             self.handle_sub_repos(distribution_tree)
 
         all_content = main_content
-        for name, content in self.sub_repos:
+        for name, content, checksums in self.sub_repos:
             os.mkdir(name)
             setattr(self, f"{name}_content", content)
+            setattr(self, f"{name}_checksums", checksums)
             setattr(self, f"{name}_repomdrecords", self.prepare_metadata_files(content, name))
             all_content |= content
         self.publish_artifacts(all_content)
 
 
-def publish(repository_version_pk, metadata_signing_service=None):
+def get_checksum(name, checksums):
```

Comment:
would you mind renaming all the occurrences from checksum to checksum_type? i find this a bit misleading because we are not storing the checksum itself but only its type.